std/digest/xxh.d

```diff
@@ -0,0 +1,3352 @@
+                cast(const(xxh_u8)*) secret, seed64);
+    if (len <= 128)
+        return XXH3_len_17to128_64b(cast(const(xxh_u8)*) input, len,
+                cast(const(xxh_u8)*) secret, secretLen, seed64);
+    if (len <= XXH3_MIDSIZE_MAX)
+        return XXH3_len_129to240_64b(cast(const(xxh_u8)*) input, len,
+                cast(const(xxh_u8)*) secret, secretLen, seed64);
+    return f_hashLong(input, len, seed64, cast(const(xxh_u8)*) secret, secretLen);
+}
+
+/* ===   Public entry point   === */
+
+/*! @ingroup XXH3_family */
+XXH64_hash_t XXH3_64bits(const(void)* input, size_t length) @trusted pure nothrow @nogc
+{
+    return XXH3_64bits_internal(input, length, 0, &XXH3_kSecret[0],
+            (XXH3_kSecret).sizeof, &XXH3_hashLong_64b_default);
+}
+
+/*! @ingroup XXH3_family */
+XXH64_hash_t XXH3_64bits_withSecret(const(void)* input, size_t length,
+        const(void)* secret, size_t secretSize) @trusted pure nothrow @nogc
+{
+    return XXH3_64bits_internal(input, length, 0, secret, secretSize,
+            &XXH3_hashLong_64b_withSecret);
+}
+
+/*! @ingroup XXH3_family */
+XXH64_hash_t XXH3_64bits_withSeed(const(void)* input, size_t length, XXH64_hash_t seed) @trusted pure nothrow @nogc
+{
+    return XXH3_64bits_internal(input, length, seed, &XXH3_kSecret[0],
+            (XXH3_kSecret).sizeof, &XXH3_hashLong_64b_withSeed);
+}
+
+XXH64_hash_t XXH3_64bits_withSecretandSeed(const(void)* input, size_t length,
+        const(void)* secret, size_t secretSize, XXH64_hash_t seed) @trusted pure nothrow @nogc
+{
+    if (length <= XXH3_MIDSIZE_MAX)
+        return XXH3_64bits_internal(input, length, seed, &XXH3_kSecret[0],
+                (XXH3_kSecret).sizeof, null);
+    return XXH3_hashLong_64b_withSecret(input, length, seed,
+            cast(const(xxh_u8)*) secret, secretSize);
+}
+
+/* ===   XXH3 streaming   === */
+/*
+ * Malloc's a pointer that is always aligned to align.
+ *
+ * This must be freed with `XXH_alignedFree()`.
+ *
+ * malloc typically guarantees 16 byte alignment on 64-bit systems and 8 byte
+ * alignment on 32-bit. This isn't enough for the 32 byte aligned loads in AVX2
+ * or on 32-bit, the 16 byte aligned loads in SSE2 and NEON.
+ *
+ * This underalignment previously caused a rather obvious crash which went
+ * completely unnoticed due to XXH3_createState() not actually being tested.
+ * Credit to RedSpah for noticing this bug.
+ *
+ * The alignment is done manually: Functions like posix_memalign or _mm_malloc
+ * are avoided: To maintain portability, we would have to write a fallback
+ * like this anyways, and besides, testing for the existence of library
+ * functions without relying on external build tools is impossible.
+ *
+ * The method is simple: Overallocate, manually align, and store the offset
+ * to the original behind the returned pointer.
+ *
+ * Align must be a power of 2 and 8 <= align <= 128.
+ */
+private void* XXH_alignedMalloc(size_t s, size_t align_) @trusted nothrow @nogc
```

Comment:
We might just use `aligned_alloc` from C stdlib.

Reply:
Yes, thanks for the hint. There is more C-ish code around, which must be fixed. E.g. all those 'base ptr + length' arguments can be changed to a referenced D array passed as argument. Also the memcpy() stuff can be changed to D syntax using assignments of array slices. But step by step ...

Reply:
Changed code to use aligned_alloc() and free() from libc.